app/config/services/repositories.yml

```diff
@@ -0,0 +1,26 @@
+services:
+  ClientRepository:
+    class: AppBundle\Entity\Client
+    factory: ["@doctrine.orm.entity_manager", getRepository]
+    arguments:
+    - AppBundle\Entity\Client
+
+  AppBundle\v2\Repository\ClientRepository:
+    class: AppBundle\v2\Repository\ClientRepository
+    arguments:
+    - '@doctrine.orm.entity_manager'
+    - '@UserRepository'
+
+  UserRepository:
+    class: AppBundle\Entity\User
+    factory: ["@doctrine.orm.entity_manager", getRepository]
+    arguments:
+      - AppBundle\Entity\User
+
+  AppBundle\v2\Repository\DeputyRepository:
+    class: AppBundle\v2\Repository\DeputyRepository
+    arguments:
+      - '@doctrine.orm.entity_manager'
+      - '@UserRepository'
+      - '@AppBundle\v2\Repository\ClientRepository'
```

Comment:
comment only: Not convinced we need to inject this. AbstractEntityRepository supplies methods to get other repositories but appreciate the need to inject to allow easier unit testing so happy to leave it in. Just conscious that as the repositories grow in complexity you may find developers adding several other repositories to the constructor as they see fit for their needs

Reply:
Unit testing is a very valid reason in my opinion. If this repository happens to need another repository, then I think it would be a valid reason to inject one.